workflow/controller/taskset.go

```diff
@@ -52,6 +52,15 @@ func (woc *wfOperationCtx) getDeleteTaskAndNodePatch() (tasksPatch map[string]in
 	}
 	return
 }
+
+func (woc *wfOperationCtx) markTaskSetNodesError(err error) {
+	for _, node := range woc.wf.Status.Nodes {
```

Comment:
Okay, so we'd set each Node that's of an Agent type to NodeError. That seems good. Just out of curiosity, do you know if it would be possible in the code for another Agent type Node to get added to `woc.wf.Status.Nodes` after we call this?

Reply:
I can confirm that each workflow will only create one agent pod to execute HTTP/Plugin templates, so all taskset nodes belong to one same agent.

Reply:
Right, I know, but I’m actually asking if a new TaskSet Node can come along and get added to woc.wf.Status.Nodes? Does my question make sense?

Reply:
I’m not saying it’s necessarily a problem if it does but good to understand expected behavior

Reply:
> I’m actually asking if a new TaskSet Node can come along and get added to woc.wf.Status.Nodes?

Sorry, I misunderstood your question. Your concerns make sense and it is possible that this could happen if onExitHandler of HTTP/Plugin types are used. Howerver, it will not cause any problems because `taskSetReconciliation` will be called again, so new Agent type nodes added will also be marked as `ERROR`.

Reply:
yes, I think this seems okay, thanks